site/docs/api/add-function-serializer.md

```diff
@@ -0,0 +1,122 @@
+---
+title: addFunctionSerializer
+parent: api
+---
+
+# addFunctionSerializer
+
+> ✋&nbsp;&nbsp;This is an advanced feature recommended for library authors only.
+
+Typically, if you try to export a function from one of your stylesheets, you'll get an error that you can only export plain objects, arrays, strings, numbers and `null`/`undefined`.
+
+If you're wanting to create higher level abstractions like [Sprinkles](../../packages/sprinkles) or [Recipes](../../packages/recipes), this is a big problem!
+
+To address this limitation, the `addFunctionSerializer` utility allows you to annotate your functions with instructions on how to serialize them.
+
+As a basic example, let's say we want to create a library called `my-awesome-styled-div` that lets you create a React component that renders a `<div>` with static CSS bound to it, and we want our consumers to use it in their `.css.ts` files like this:
+
+```ts
+// MyComponent.css.ts
+
+import { styledDiv } from 'my-awesome-styled-div';
+
+export const MyComponent = styledDiv({
+  background: 'blue',
+  padding: 12
+});
+```
+
+Normally if you tried to do this, since `MyComponent` is a function, it would trigger an error during the build since it can't be serialized.
+
+Luckily `addFunctionSerializer` can help us!
+
+To understand how we could make this API work, we'll first look at what you might expect the implementation of `styledDiv` to look like.
+
+```tsx
+// index.css.tsx
+
+import React from 'react';
+import { style } from '@vanilla-extract/css';
+
+export function styledDiv(styles) {
+  const className = style(styles);
+
+  return function Component(
+    props: React.ComponentProps<'div'>
+  ) {
+    return (
+      <div
+        {...props}
+        className={[props.className, className]
+          .filter(Boolean)
+          .join(' ')}
+      />
+    );
+  };
+}
+```
+
+This is a pretty simple implementation, but it has a couple of major problems that prevent it from working. Firstly, there's no way to convert this function to a string so it can be added to your runtime JavaScript bundle. Secondly, even if it could be serialized somehow, it relies on `@vanilla-extract/css` which can't be used in the browser.
```

Comment:
As discussed, could we re-word this a bit as technically `@vanilla-extract/css` can be used in a browser?